File: djyosof/audio_types/spotify_track.py
import discord

from djyosof.audio_types.playable_audio import AudioType, PlayableAudio


class SpotifyTrack(PlayableAudio):
    def __init__(self, search_response: dict):
        self.name = search_response["name"]
        self.artist = search_response["artists"][0]["name"]
        self.album = search_response["album"]["name"]
        self.album_art_url = search_response["album"]["images"][1]["url"]
        self.track_id = search_response["id"]
        self.audio_stream = None

    def get_embed(self):
        embed = discord.Embed(
            title="Now Playing",
            color=discord.Colour.blurple(),
        )
        embed.add_field(name="Track", value=self.name, inline=True)
        embed.add_field(name="Artist", value=self.artist, inline=True)
        embed.add_field(name="Album", value=self.album, inline=True)
        embed.set_image(url=self.album_art_url)
        return embed

    def get_type(self):
        return AudioType.spotify

File: audio/spotify_track.go
package audio

import (
	"fmt"

	"github.com/disgoorg/disgo/discord"
)

// SpotifyTrack holds the metadata needed to display and stream a Spotify track.
// Mirrors djyosof/audio_types/spotify.py.
type SpotifyTrack struct {
	Name        string
	Artist      string
	Album       string
	AlbumArtURL string
	// TrackID is the base62 Spotify track id (the "{22}" portion of a link).
	TrackID string
}

// Embed implements PlayableAudio.
func (t *SpotifyTrack) Embed() discord.Embed {
	return discord.NewEmbed().
		WithTitle("Now Playing").
		AddField("Track", t.Name, false).
		AddField("Artist", t.Artist, false).
		AddField("Album", t.Album, false).
		WithImage(t.AlbumArtURL)
}

// Type implements PlayableAudio.
func (t *SpotifyTrack) Type() Type { return TypeSpotify }

// DisplayName implements PlayableAudio.
func (t *SpotifyTrack) DisplayName() string {
	return fmt.Sprintf("%s by %s", t.Name, t.Artist)
}
